doctoshotgun.py

```diff
@@ -423,7 +431,20 @@ def main(self):
 
         log('Starting to look for vaccine slots for %s %s...', docto.patient['first_name'], docto.patient['last_name'])
         log('This may take a few minutes/hours, be patient!')
-        cities = [docto.normalize(city) for city in args.city.split(',')]
+        input_cities = [docto.normalize(city) for city in args.city.split(',')]
+        cities = []
+        check_zip = {}
+
+        for city in input_cities :
+            reg_zipcode = re.match('([0-9]*)-([a-z]*)', city)
+            if reg_zipcode : 
+                cities.append(docto.normalize(reg_zipcode[2]))
+                if docto.normalize(reg_zipcode[2]) in check_zip:
+                    check_zip[docto.normalize(reg_zipcode[2])].append(reg_zipcode[1])
+                else:
+                    check_zip[docto.normalize(reg_zipcode[2])] = [reg_zipcode[1]]
+            else :
+                cities.append(docto.normalize(city))
```

Comment:
Why all this normalizing on an already normalized string?

Also I think the regex needs to be improved to assert valid input, and named capture groups can make the code more readable.
```suggestion
        reg_city_zipcode = re.match('^((?P<zip>[0-9]+)-)?(?P<city>[a-z][a-z-]*)$', city)
        if not reg_city_zipcode:
            log('Invalid city parameter.', color='red')
            return 1
        extracted_city = reg_city_zipcode['city']
        extracted_zip = reg_city_zipcode['zip']
        if not extracted_city in cities:
            cities.append(extracted_city)
        if extracted_zip:
            if extracted_city in check_zip:
                check_zip[extracted_city].append(extracted_zip)
            else:
                check_zip[extracted_city] = [extracted_zip]
```